en/engine/event/event-emit.md

```diff
@@ -0,0 +1,150 @@
+this.node.on(Node.EventType.MOUSE_DOWN, function ( event ) {
+  this.enabled = false;
+}.bind(this));
+
+// Using the third parameter
+this.node.on(Node.EventType.MOUSE_DOWN, (event) => {
+  this.enabled = false;
+}, this);
+```
+
+Besides listening with `on`, we can also use the `once` method. The `once` listener will shut the event being listened to after the listener function responds.
+
+## Shut listener
+
+We can shut the corresponding event listener using `off` when we don't care about a certain event anymore.
+
+The `off` method can be used in two ways
+
+```ts
+// Cancel all registered events of this type on the object.
+xxx.off(type);
+// Cancels events on objects with this type of callback designation target.
+xxx.off(type, func, target);
+```
+
+One thing to note is that the parameter of `off` must be in one-to-one correspondence with the parameter of `on` in order to shut it.
+
+Below are what we recommend you to put in:
+
+```ts
+import { _decorator, Component, Node } from "cc";
+const { ccclass } = _decorator;
+
+@ccclass("Example")
+export class Example extends Component {
+    onEnable(){
+        this.node.on('foobar', this._sayHello, this);
+    }
+
+    onDisable(){
+        this.node.off('foobar', this._sayHello, this);
+    }
+
+    _sayHello(){
+        console.log('Hello World');
+    }
+}
+```
+
+## Event Dispatches
+
+There are two ways to trigger an event: `emit` and `dispatchEvent`. The difference between the two is that the latter can do event passing. Let's start with a simple example of the `emit` event.
+
+```ts
+// At most 5 args could be emit.
+xxx.emit(type, ...args);
+```
+
+## Explanation for event arguments
+
+When emitting event, We can start passing our event parameters at the second argument of the emit function. At the same time, the corresponding event parameters are available in the `on` registered callback.
+
+```ts
+import { _decorator, Component, Node } from "cc";
+const { ccclass } = _decorator;
+
+@ccclass("Example")
+export class Example extends Component {
+    onLoad () {
+        this.node.on('foo', (arg1, arg2, arg3) => {
+            console.log(arg1, arg2, arg3);  // print 1, 2, 3
+        });
+    }
+
+    start () {
+        let arg1 = 1, arg2 = 2, arg3 = 3;
+        // At most 5 args could be emit.
+        this.node.emit('foo', arg1, arg2, arg3);
+  }
+}
+```
+
+Note that only up to 5 event parameters can be passed here for the performance of the underlying event distribution. Therefore, care should be taken to control the number of parameters passed when passing a parameter.
+
+## Event delivery
+
+Events launched by the `dispatchEvent` method mentioned above would enter the event delivery stage. In Cocos Creator's event delivery system, we use bubble delivery. Bubble delivery will pass the event from the initiating node continually on to its parent node,  until the root node is reached or an interrupt `event.propagationStopped = true` is made in the response function of a node.
```

Comment:
In Cocos Creator's event delivery system, we use bubble delivery.
-->
In Cocos Creator's event delivery system, bubble delivery is used.